scala/lambdas/entity-event-generator-lambda/src/main/scala/uk/gov/nationalarchives/entityeventgenerator/Lambda.scala

```diff
@@ -62,41 +58,46 @@ class Lambda extends LambdaRunner[ScheduledEvent, Int, Config, Dependencies] {
       entitiesClient: EntityClient[IO, Fs2Streams[IO]],
       dADynamoDBClient: DADynamoDBClient[IO],
       dASnsDBClient: DASNSClient[IO],
-      startFrom: Int,
       eventTriggeredDatetime: OffsetDateTime
-  ): IO[Int] =
+  ): IO[Int] = {
+    val ignoredEventTypes = List("Download", "Characterise", "VirusCheck")
     for {
       updatedSinceResponses <- dADynamoDBClient.getItems[GetItemsResponse, PartitionKey](
         List(PartitionKey("LastPolled")),
         config.lastEventActionTableName
       )
       updatedSinceResponse = updatedSinceResponses.head
       updatedSinceAsDate = OffsetDateTime.parse(updatedSinceResponse.datetime).toZonedDateTime
-      recentlyUpdatedEntities <- entitiesClient.entitiesUpdatedSince(updatedSinceAsDate, startFrom)
+      currentStart = updatedSinceResponse.start
+      recentlyUpdatedEntities <- entitiesClient.entitiesUpdatedSince(updatedSinceAsDate, currentStart)
       _ <- logger.info(s"There were ${recentlyUpdatedEntities.length} entities updated since $updatedSinceAsDate")
 
       entityLastEventActionDate <-
         if (recentlyUpdatedEntities.nonEmpty) {
           val lastUpdatedEntity: Entity = recentlyUpdatedEntities.last
           entitiesClient.entityEventActions(lastUpdatedEntity).map { entityEventActions =>
-            Some(entityEventActions.head.dateOfEvent.toOffsetDateTime)
+            Some(entityEventActions.filterNot(ev => ignoredEventTypes.contains(ev.eventType)).head.dateOfEvent.toOffsetDateTime)
           }
         } else IO.pure(None)
 
       _ <- IO.whenA(entityLastEventActionDate.exists(_.isBefore(eventTriggeredDatetime))) {
         for {
           _ <- dASnsDBClient.publish[CompactEntity](config.snsArn)(convertToCompactEntities(recentlyUpdatedEntities.toList))
           updateDateAttributeValue = AttributeValue.builder().s(entityLastEventActionDate.get.toString).build()
+          // This is to cover the case where there are more than 1000 items with the same last event action date
+          start = if entityLastEventActionDate.get.isEqual(OffsetDateTime.parse(updatedSinceResponse.datetime)) then currentStart + 1000 else 0
```

Comment:
Maybe `currentResultNum`, `currentPosition`,  `nextStart`, `startNext`, `nextStartPoint`, `nextStartingPoint`?

Reply:
`nextStart` is ok.